packages/vite/src/node/utils.ts

```diff
@@ -534,6 +534,34 @@ export function removeDirSync(dir: string) {
   }
 }
 
+const REMOVE_DIR_TIMEOUT = 5000
+const rmdir = promisify(fs.rm ?? fs.rmdir) // TODO: Remove after support for Node 12 is dropped
+export async function removeDir(dir: string) {
+  const t0 = Date.now()
+  const attempt = async () => {
+    try {
+      await rmdir(dir, { recursive: true })
+    } catch (e) {
+      if (e.code === 'ENOENT') {
+        return
+      }
+      if (e.code === 'ENOTEMPTY' || e.code === 'EPERM') {
+        if (Date.now() - t0 < REMOVE_DIR_TIMEOUT) {
+          // There's no delay here, testing showed it wasn't required.
+          // If the remove doesn't succeed the first time, it almost always succeed immediately afterwards.
+          // In rare cases two or three attempts are needed before the dir is successfully removed,
+          // typically taking <1s total.
+          await attempt()
+          return
```

Comment:
If there is a real issue, this will end up generating tons of calls though. I think we need the same backoff scheme that is used in `renameDir`. As a note, I wonder if the race condition is because someone is reading the files when we try to delete them, but in that case, retrying seems to be the simplest solution.

Reply:
I've added a backoff delay. Thought it'd be best to take the FS time into account - no point adding time between operations that already take time to complete.